flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -59,6 +59,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
           Literal(str.toInt)
         } else if (str.endsWith("f") | str.endsWith("F")) {
           Literal(str.toFloat)
+        } else if (str.endsWith(".")) {
```

Comment:
So in that sense what is the valid number expression?  I cannot say 1.CAST(STRING) then?
If that is not valid then 1.abs() is also not vaild right?